src/python/gudhi/representations/vector_methods.py

```diff
@@ -350,6 +351,92 @@ def __call__(self, diag):
         """
         return self.fit_transform([diag])[0,:]
 
+
+class IrregularBettiCurve(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing Betti curves from a list of persistence diagrams. Unlike the BettiCurve class, the curves are not sampled at fixed, regular points. Each curve has all its data intact.
+    """
+
+    def __init__(self, filtration_range = [np.nan, np.nan]):
+        """
+        Constructor for the IrregularBettiCurve class.
+
+        Parameters:
+            filtration_range ([float, float]): minimum and maximum filtration range to consider, of the form [x_min, x_max] (default [numpy.nan, numpy.nan]). The behavior is not the same as for sample_range in BettiCurve. In particular, intervals born before x_min are not considered. Intervals born in [x_min, x_max) contribute to the Betti curves as if they were infinite.
+        """
+        self.filtration_range = np.array(filtration_range)
+
+    def fit(self, X, y = None):
+        """
+        See the corresponding function in BettiCurve.
+        """
+        if np.isnan(self.filtration_range).any():
```

Comment:
If any NaN, `fit` will do nothing, but `transform` will assert. This is an open question also for @MathieuCarriere , but I am not sure if we should homogenize the behaviour (always assert or always do nothing, or even an exception).

Reply:
I am more in favour of homogeneous behavior, so I would suggest no assert, but I am ok to keep it this way